aspnetcore/fundamentals/http-requests.md

```diff
@@ -0,0 +1,360 @@
+        var result = await _httpClient.GetAsync("api/values");
+        var resultObj = Enumerable.Empty<string>();
+
+        if (result.IsSuccessStatusCode)
+        {
+            resultObj = JsonConvert.DeserializeObject<IEnumerable<string>>(await result.Content.ReadAsStringAsync());
+            _cache.Set("GetValue", resultObj);
+        }
+        else
+        {
+            if (_cache.TryGetValue("GetValue", out resultObj))
+            {
+                _logger.LogWarning("Returning cached values as the values service is unavailable.");
+                return resultObj;
+            }
+            result.EnsureSuccessStatusCode();
+        }        
+        return resultObj;
+    }
+}
+```
+
+In this example, the `HttpClient` is stored as a private field and all access to make external calls goes through the GetValues method.
+
+## Generated clients
+
+`HttpClientFactory` can be used in combination with other third-party libraries such as [Refit](https://github.com/paulcbetts/refit). Refit is a REST library for .NET that is inspired by Square's [Retrofit](http://square.github.io/retrofit/) library. It turns your REST API into a live interface. This interface defines a external REST API. An implementation of the interface is generated by the `RestService` using `HttpClient` to make its calls. 
+
+First an interface and reply are defined which represent the exteranl API and its response.
+
+```csharp
+public interface IHelloClient
+{
+    [Get("/helloworld")]
+    Task<Reply> GetMessageAsync();
+}
+
+public class Reply
+{
+    public string Message { get; set; }
+}
+```
+
+A typed client can then be added, using Refit to generate the implementation.
+
+```csharp
+public void ConfigureServices(IServiceCollection services)
+{
+    services.AddHttpClient("hello", c =>
+    {
+        c.BaseAddress = new Uri("http://localhost:5000");
+    })
+    .AddTypedClient(c => Refit.RestService.For<IHelloClient>(c));
+
+    services.AddMvc();
+}
+```
+
+The defined interface can then be consumed wherever necessary in your code, with the implementation provided by the DI framework and Refit.
+
+```csharp
+[ApiController]
+public class ValuesController : ControllerBase
+{
+    private readonly IHelloClient _client;
+
+    public ValuesController(IHelloClient client)
+    {
+        _client = client;
+    }
+    
+    [HttpGet("/")]
+    public async Task<ActionResult<Reply>> Index()
+    {
+        return await _client.GetMessageAsync();
+    }
+}
+```
+
+## Outgoing request middleware
+
+The `HttpClientFactory` supports registering and chaining `DelegatingHandlers` to easily build an outgoing request middleware pipeline. Each of these handlers is able to perform work before and after the outgoing request, in a very similar pattern to the middleware pipeline in ASP.NET Core. This provides a mechanism to manage cross cutting concerns around the requests an app is making. This includes things such as caching, error handling, serialization and logging.
```

Comment:
@scottaddie Do you have a preference around the code backticks around pluralised types? Inclusive of the "s" or not?

Reply:
I try to avoid the plural form in backticks, as it looks awkward. I suggest rewriting to something like the following:
The `HttpClientFactory` supports registering and chaining a `DelegatingHandler` to easily build an outgoing request middleware pipeline.